### PR TITLE
Font optimizations

### PR DESCRIPTION
**Pull request by zachleat**
* Removes old formats that aren’t needed anymore (WOFF2/WOFF are most important, TTF for old androids)
* Adds WOFF2 saving a bunch of KB
* Adds zopfli encoding to WOFF to save a few KB
* Subsets both fonts saving a bunch of KB
* Uses font-display: swap to get rid of FOIT (in Chrome, soon in FF)
* Uses preload to make fonts fetch and render faster (in Chrome, soon in FF)
* Updates serviceworker to use WOFF2 only (all browsers that support sw support woff2)

### CODE REVIEW

**Comment by jennschiffer** [comment]
u rule 🎅

**Comment by zachleat** [comment]
if you ever have to make changes to this later, I documented what I did here https://www.zachleat.com/web/23-minutes/

File: index.html [hunk 1]
```diff
@@ -3,6 +3,8 @@
 <head>
   <meta charset="UTF-8">
   <title>make 8-bit art!</title>
+  <link rel="preload" href="/assets/fonts/8bit-Art-Sans-subset.woff2" as="font" type="font/woff2" crossorigin>
+  <link rel="preload" href="/assets/fonts/VT323-Regular-subset.woff2" as="font" type="font/woff2" crossorigin>
```
Comment (CrocoDillon) [inline index.html]: Out of curiosity, what does the `crossorigin` attribute do (since it's a same origin request)?

Comment (zachleat) [reply index.html]: > One point worth going over: You have to add a crossorigin attribute when fetching fonts, as they are fetched using anonymous mode CORS. Yes, even if your fonts are on the same origin as the page. Sorry.

via https://www.smashingmagazine.com/2016/02/preload-what-is-it-good-for/

Comment (CrocoDillon) [reply index.html]: Interesting, thanks for the clarification 😄

**Pull request merged**